modules/openthread/Kconfig

```diff
@@ -162,48 +162,81 @@ config OPENTHREAD_SECURITY_CUSTOM_CONFIG
 
 config OPENTHREAD_SECURITY_DEFAULT_CONFIG
 	bool "Default configuration"
-	select OPENTHREAD_MBEDTLS_CONFIG
 	help
 	  Use the OpenThread Mbed TLS configuration pre-defined security scheme.
 
 endchoice # OPENTHREAD_SECURITY
 
-config OPENTHREAD_MBEDTLS_CONFIG
-	bool
-	select MBEDTLS
-	select MBEDTLS_ENABLE_HEAP
+if OPENTHREAD_SECURITY_DEFAULT_CONFIG
+
+choice OPENTHREAD_CRYPTO_IMPLEMENTATION
```

Comment:
Previously, enabling PSA crypto backend didn't depend on using `OPENTHREAD_SECURITY_DEFAULT_CONFIG` which was more flexible because one could enable PSA crypto backend without the default config (e.g. a user may have a custom PSA crypto implementation that uses other configuration scheme). Any reason this needs to be coupled now?